File: ws/ws22UnitTesting/BasicOperations/src/main/java/ec/edu/espe/basicoperations/view/BasicOperations.java
package ec.edu.espe.basicoperations.view;

import ec.edu.espe.basicoperations.controller.BasicOperation;
import ec.edu.espe.basicoperations.controller.PersonController;
import java.time.LocalDate;

/**
 *
 * @author G400
 */
public class BasicOperations {

    public static void main(String[] args) {
        System.out.println("Basic Operations! \nKerlly Chiriboga, OOP 14539");
        System.out.println("---> Unit Testing in action <---");

        float addend1;
        float addend2;
        float sum;


        addend1 = 1.2F;
        addend2 = 2.4F;

        sum = addend1 + addend2;

        System.out.println("The sum of " + addend1 + " + " + addend2 + "--> " + sum);


        addend1 = 1.8F;
        addend2 = -2.2F;        
        
        
        sum = BasicOperation.addTwoNumbers(addend1, addend2);
        
        System.out.println("The sum of " + addend1 + " + " + addend2 + "--> " + sum);

        PersonController personController = new PersonController();
        
        LocalDate bornOnDate;
        bornOnDate = LocalDate.parse("2004-01-13");
        int days;
        
        days = personController.computageAgeInDays(bornOnDate);
        
        System.out.println("My age in days is --> " + days);
        
        int numberOfYears;
        numberOfYears = personController.computageAgeInYears(bornOnDate);
        System.out.println("My age in years is --> " + numberOfYears);
        
        
    }

}

File: hw/u2/hw15TestCasesJavaScript/src/views/BasicOperations.mjs

const BasicOperation = require('../controllers/BasicOperation');
const PersonController = require('../controllers/PersonController');

console.log("Basic Operations! \nKerlly Chiriboga, OOP 14539");
console.log("---> Unit Testing in action <---");

let addend1;
let addend2;
let sum;

addend1 = 1.2;
addend2 = 2.4;
sum = addend1 + addend2;
console.log(`The sum of ${addend1} + ${addend2} --> ${sum}`);

addend1 = 1.8;
addend2 = -2.2;
sum = BasicOperation.addTwoNumbers(addend1, addend2);
console.log(`The sum of ${addend1} + ${addend2} --> ${sum}`);

const personController = new PersonController();
const birthDate = '2004-01-13';

const days = personController.computageAgeInDays(birthDate);
console.log(`My age in days is --> ${days}`);

const years = personController.computageAgeInYears(birthDate);
console.log(`My age in years is --> ${years}`);
